playwright.config.ts

```diff
@@ -0,0 +1,50 @@
+import { PlaywrightTestConfig, devices } from '@playwright/test';
+
+const config: PlaywrightTestConfig = {
+  use: {
+    // Run the browser in headless mode
+    // headless: true,
+    // Record trace for each test, but remove it from successful test runs.
+    trace: 'retain-on-failure',
+  },
+  // Look for test files in the "test/rendering" directory, relative to this configuration file
+  testDir: 'test/rendering',
+  testMatch: /.*\.render\.ts/,
+  outputDir: './test/rendering/artifacts/',
+  reporter: [
+    ['list'],
+    [
+      'html',
+      {
+        outputFolder: './test/rendering/test-report',
+        open: process.env.CI ? 'never' : 'on-failure',
+      },
+    ],
+  ],
+  // Forbid test.only on CI
+  forbidOnly: !!process.env.CI,
+  // The maximum number of retry attempts per test, two retries on CI
+  retries: process.env.CI ? 2 : 0,
+  // Limit the number of workers
+  workers: 1,
+  // Multiple "projects" can run your tests in multiple browsers and configurations
+  projects: [
+    {
+      name: 'chromium',
+      use: { ...devices['Desktop Chrome'] },
+    },
+    {
+      name: 'firefox',
+      use: { ...devices['Desktop Firefox'] },
+    },
+    // {
+    //   name: 'webkit',
+    //   use: { ...devices['Desktop Safari'] },
+    // },
```

Comment:
why firefox and chrome but not webkit?

Reply:
I am not able to let circle CI handle webkit for now. It lacks system dependencies but did not find related docs.

Reply:
ok then skip webkit for now